src/tcp_sock.rs

```diff
@@ -45,6 +43,26 @@ impl TcpSock {
         }
     }
 
+    /// Specify data encryption context which will determine how outgoing data is encrypted.
+    pub fn use_encrypt_ctx(&mut self, enc_ctx: EncryptContext) -> ::Res<()> {
```

Comment:
use `set_` to be more rust-like (also used in other setters in our libs i think)